src/main/java/com/movie_collection/dal/dao/MovieDAO.java

```diff
@@ -74,9 +80,44 @@ public List<Movie> getAllMovies() throws SQLException {
                 double rating = rs.getDouble("rating");
                 StringProperty path = new SimpleStringProperty(rs.getString("path"));
                 Date lastview = rs.getDate("lastview");
-                movies.add(new Movie(id, name, rating, path, lastview));
+                movies.add(new Movie(id, name, rating, path, getCategoriesOfMovie(id), lastview));
             }
         }
         return (movies);
     }
+
+    public List<Movie> getAllMoviesInTheCategory(int categoryId) throws SQLException {
+        ArrayList<Movie> movies = new ArrayList<>();
+        try (Connection connection = cm.getConnection()) {
+            String sql = "SELECT * FROM Movie INNER JOIN CatMovie ON Movie.id = CatMovie.MovieId WHERE CatMovie.CategoryId = ?";
+            PreparedStatement preparedStatement = connection.prepareStatement(sql);
+            preparedStatement.setInt(1, categoryId);
+            ResultSet rs = preparedStatement.executeQuery();
+            while (rs.next()) {
+                int id = rs.getInt("id");
+                StringProperty title = new SimpleStringProperty(rs.getString("title"));
+                double rating = rs.getDouble("imdbRating");
+                StringProperty path = new SimpleStringProperty(rs.getString("path"));
+                Date lastview = rs.getDate("lastview");
+                movies.add(new Movie(id, title, rating, path, getCategoriesOfMovie(id), lastview));
+            }
+        }
+        return movies;
+    }
+
+    public List<Category> getCategoriesOfMovie(int MovieId) throws SQLException {
+        ArrayList<Category> allCategories = new ArrayList<>();
+        try (Connection con = cm.getConnection()) {
+            String sql = "SELECT * FROM Category JOIN CatMovie ON Category.id = CatMovie.CategoryId WHERE MovieId = ?";
```

Comment:
can we not use * in the select and rather id, title etc 
its only for practise again since some other programme can go after u work on the code and he has no idea without looking up the database what fields u are selecting 
obviously there are ways that u can even hide all the fields but for now I would stick with explicit naming of all  ?

Reply:
https://github.com/TomassSimko/Private-Movie-Collection/pull/9/commits/ef7263ec26617d666f223e684bd1a8de326bd19e

Reply:
Great recommendation - will do so in the future
Fixed here
https://github.com/TomassSimko/Private-Movie-Collection/pull/9/commits/26443fe74d07f2cc9da95e2c9549c68b0dc14daa